src/eda_server/batch.py

```diff
@@ -0,0 +1,34 @@
+import asyncio
+import logging
+import time
+
+logger = logging.getLogger("eda_server.batch")
+
+
+class Batcher:
+    def __init__(self, processor_fn, max_batch_size=10000, batch_timeout=1):
+        self.queue = asyncio.Queue()
+        self.max_batch_size = max_batch_size
+        self.batch_timeout = batch_timeout
+        self.processor_fn = processor_fn
+
+    def start(self, get_db_session_factory):
+        self.get_db_session_factory = get_db_session_factory
+        _ = asyncio.get_event_loop()
+        asyncio.create_task(self._batcher())
+
+    async def _batcher(self):
```

Comment:
The logic here looks not very intuitive and probably not performant as well.
It runs a loop to check the queue size with fixed intervals and then passes the queue to the function that consumes messages from it. The queue size is unlimited which may lead to the back pressure problem.

While this somehow works because queue size is known and the queue access doesn't need synchronization, I'd re-write it in a different manner.

1. Use limited queue size to avoid uncontrolled growth of a queue.
2. The `processor_fn` callback should expect full batch: a sequence (e.g. list) of items to be processed.
2. The Batcher consumes the queue with a timeout:

```
# PSEUDOCODE #

while True:
    try:
        item = await asyncio.wait(queue.get(), timeout - elapsed_time)
    except TimeoutError:
        # flush
    batch.append(item)
    if len(batch) > max_batch_size:
        # flush
```